### PR TITLE
fix: Category Aware Search Profile not applied on Category Page (#5510)

### LINKED ISSUE #5510: Adaptive Search Profile is not applied on Category Page
## Environment Details

- Spartacus:  1.2.0

## Steps to Reproduce

1. Prepare Category Aware Search Profile (to be done in Adaptive Search) with some configuration e.g promoted products.
2. Make sure configuration is synchronized to online catalog version.
3. Navigate to category with adaptive search changes applied.
4. Search Profile changes for category is not applied.

## Observed Results
Requests sent from spartacus does not narrow down search to category and rather treat category as facet.

## Expected Results
Category Aware Search Profile changes are applied Category Page

## Offer to Fix
No

## Additional Information
Simple Search Profile works fine. Problem is only occurring for Category Aware Search Profile.
On old storefront it works fine.

### PR DESCRIPTION
**Pull request by neilhubertprice**
Replacement PR for [https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/5909] this time for version 2.0, as requested by @tobi-or-not-tobi 

When using Adaptive Search only Global profiles were applying, but not
category profiles. This is because Adaptive Search expects the use of
the allCategories field (either as a Filter Query or a Facet Value) to
mark a category page query.

This affects both categories and brands (previouly using the category
and brand facets respectively). Both have been changed to query on
the allCategories field.

The JSP storefronts all use the allCategories field for all category
page requests & do not differentiate between category/brand/etc when
querying.
Also fixed CategoryPageMetaResolver to understand the allCategories facet.

Note: Commerce setup needs modifying for this to work correctly. The
allCategories facet by default has no Facet Value Display Name Provider
set. Without that the breadcrumb displays the code rather than the name.
Set this to the same as the category facet (categoryFacetDisplayNameProvider)
to have proper names in the breadcrumb

closes #5510

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.ts [hunk 1]
```diff
@@ -37,8 +37,8 @@ export class ProductListComponentService {
 
   protected sub: Subscription;
 
-  protected readonly RELEVANCE_CATEGORY = ':relevance:category:';
-  protected readonly RELEVANCE_BRAND = ':relevance:brand:';
+  protected readonly RELEVANCE_CATEGORY = ':relevance:allCategories:';
+  protected readonly RELEVANCE_BRAND = ':relevance:allCategories:';
```
Comment (tobi-or-not-tobi) [inline projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.ts]: we should get ride of this specific constant, as we rely now on a single SOLR index property.

File: projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.spec.ts [hunk 2]
```diff
@@ -132,7 +132,7 @@ describe('ProductListComponentService', () => {
         service.model$.subscribe();
 
         expect(productSearchService.search).toHaveBeenCalledWith(
-          ':relevance:category:testCategory',
+          ':relevance:allCategories:testCategory',
```
Comment (tobi-or-not-tobi) [inline projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.spec.ts]: we should drop the brand related test

**Comment by tobi-or-not-tobi** [comment]
There are more brand-related aspects to this code, but we can follow up in separate tickets to remove that from the code base.

File: projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.spec.ts (hunk 2 above)
Comment (neilhubertprice) [reply projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.spec.ts]: We still have the brand route until/unless other changes happen. This test is for the scenario where we have brandCode

File: projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.ts (hunk 1 above)
Comment (neilhubertprice) [reply projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.ts]: Change committed

**Pull request merged**